docs/pages/user-guides/testing/specflow.md

```diff
@@ -0,0 +1,194 @@
+﻿---
+title: Testing with SpecFlow
+layout: single
+permalink: /user-guides/testing-with-specflow/
+sidebar:
+  nav: "user-guides"
+toc: true
+---
+
+[SpecFlow](https://specflow.org/) is the most popular Behavior-Driven Development (BDD) test framework for .NET.
+In SpecFlow, tests are written as Given-When-Then scenarios in Gherkin feature files.
+Then, each line of Gherkin is automated via a step definition method written in a .NET programming language like C#.
+BDD test frameworks like SpecFlow separate test *cases* from test *code*.
+
+Boa Constrictor works together seamlessly with SpecFlow.
+In fact, Boa Constrictor was initially developed as part of a large test automation project that used SpecFlow as its core framework!
+With Boa Constrictor, SpecFlow's step definitions become very concise, readable, and safe.
+
+This user guide shows how to integrate Boa Constrictor with SpecFlow.
+The test case in the example code below will follow the same steps
+as the test case from the [tutorial]({{ "/tutorial/overview/" | relative_url }}):
+a Web UI test for DuckDuckGo searches.
+
+
+## SpecFlow test projects
+
+Boa Constrictor can integrate with any SpecFlow test project.
+If you are new to SpecFlow, read [SpecFlow's official documentation](https://docs.specflow.org/projects/specflow/en/latest/)
+to learn how to get started with SpecFlow.
+Steps to set up a SpecFlow test project include:
+
+1. Install the [SpecFlow extension for Visual Studio](https://docs.specflow.org/projects/specflow/en/latest/visualstudio/visual-studio-installation.html).
+2. Create a new [SpecFlow project in Visual Studio](https://docs.specflow.org/projects/specflow/en/latest/Installation/Project-and-Item-Templates.html).
+3. Install the following NuGet packages for the project:
+   - [Boa.Constrictor](https://www.nuget.org/packages/Boa.Constrictor/)
+   - [SpecFlow](https://www.nuget.org/packages/SpecFlow/)
+
+You may need to add other NuGet packages like
+[FluentAssertions](https://www.nuget.org/packages/FluentAssertions/) as well.
+
+To run SpecFlow tests, you will also need to choose a
```

Comment:
Something to consider - if a user has followed step 2 (above) to create a new project using the specflow project template, is it redundant to instruct a user to install the SpecFlow nuget package (step 3 above) and a unit test provider (this paragraph)? Those package references will be added by the specflow project template.